content/news/2023-06-from-single-files-to-collections/index.md

```diff
@@ -0,0 +1,180 @@
+Therefore, it should be a peace of cake to port complete workflows that 
+were based on processing single files to use collections as well.
+
+However, when applying this idea on our latest metagenomics workflow  [Foodborne Pathogen detection](https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html) we encountered some problems 
+that arise when switching from single files to collections. 
+In the following we would like to present some of those issues and how we solved them, in the hopes that these strategies can help
+others to port their workflows to collections as well.
+
+# Case 1 - Simple inputs for workflow logic
+
+It is often useful to add simple inputs to a workflow such as integers or text to specify specific parameters of tools. Galaxy can also use the output of a tool as input
+parameter for another tool. Details are described in the tutorial [Using Workflow Parameters](https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/workflow-parameters/tutorial.html). In the case of the *Single File Foodborne Pathogen Detection Workflow* a text input `Sample ID` is used downstream by multiple tools as input.
+
+The figure below shows 1 out of 5 sub-workflows, which form together the complete pathogen detection workflow. This sub-workflow is the *Gene based pathogenic identification workflow*, which takes as an input the pre-processed sample reads and the `Sample ID` both marked in green, and detects possible pathogens by identifying virulence factors (VFs) for all contigs of the input sample. 
+
+![Gene Based Pathogenic Identification For Single Sample Workflow](./figs/genebasedWF_single_marked.png)
+
+</div>  
+<figcaption>
+  Gene based pathogenic identification workflow for single files: Green boxes identifies the single sample input sequences file and the text Sample ID. These inputs needed to be changed to allow  processing of collections. 
+</figcaption>
+</div>
+
+
+This workflow could not be transformed straight forward into a collection logic, since the input in this case would need to be a list with matching `Sample IDs`.
+The problem was solved by using the name of the collection elements as `Sample IDs` and transforming them into workflow parameters using the following set of tools marked in Green, which are [Extract element identifiers](https://usegalaxy.eu/?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fcollection_element_identifiers%2Fcollection_element_identifiers%2F0.0.2&version=0.0.2), [Split file](https://usegalaxy.eu/?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Fsplit_file_to_collection%2Fsplit_file_to_collection%2F0.5.0&version=0.5.0) and `Parse parameter value` (only available in the workflow editor). A [PR](https://github.com/galaxyproject/iwc/pull/182) to add the workflow to IWC is currently pending in progress.
```

Comment:
That's not the right IWC PR

Reply:
Sorry, but that is the PR for the collection based foodborne WF: https://github.com/galaxyproject/iwc/pull/182/files#diff-4a26809d9fce447a2d5a8d192a6d3fcaf259b0bde726035cb35173bcffa39e41R124-R142 ?

Reply:
oops, didn't paste the whole link and went to https://github.com/galaxyproject/iwc/pull/18 ...